va/va_cp.h

```diff
@@ -0,0 +1,420 @@
+            uint32_t ref_frame_idx1                       : 3;
+            /** \brief Corresponds to ref_frame_sign_bias[GOLDEN_FRAME] 
+              * as the VP9 specification */
+            uint32_t ref_frame_sign_bias1                 : 1;
+            /** \brief Corresponds to ref_frame_idx[2] 
+              * as the VP9 specification */
+            uint32_t ref_frame_idx2                       : 3;
+            /** \brief Corresponds to ref_frame_sign_bias[ALTREF_FRAME] 
+              * as the VP9 specification */
+            uint32_t ref_frame_sign_bias2                 : 1;
+            /** \brief Same as the VP9 bitstream syntax element. */
+            uint32_t frame_parallel_decoding_mode         : 1;
+            /** \brief Same as the VP9 bitstream syntax element. */
+            uint32_t render_and_frame_size_different      : 1;
+            /** \brief Reserved for future use, must be zero */
+            uint32_t reserved                             : 1;
+            } bits;
+        uint32_t value;
+    } header_fields; 
+     /** \brief Same as the VP9 bitstream syntax element. */
+    uint16_t frame_width_minus1;
+     /** \brief Same as the VP9 bitstream syntax element. */
+    uint16_t frame_height_minus1;
+    /** \brief Same as the VP9 bitstream syntax element. */
+    uint16_t render_width_minus1;
+    /** \brief Same as the VP9 bitstream syntax element. */
+    uint16_t render_height_minus1;
+    /** \brief Same as the VP9 bitstream syntax element. */
+    uint8_t refresh_frame_flags;
+    /** \brief Reserved bytes for future use, must be zero */
+    uint32_t va_reserved[VA_PADDING_MEDIUM];
+} VACencSliceParameterBufferVP9;
+
+/** \brief Cenc Slice Buffer Type*/
+typedef enum {
+    /** \brief Parsed slice parameters \c VACencSliceParameterBuffer* */
+    VaCencSliceBufParamter = 1,
+    /** \brief Raw slice header of bitstream*/
+    VaCencSliceBufRaw      = 2
+}VACencSliceBufType;
+
+/** \brief Buffer for CENC status reporting*/
+typedef struct _VACencStatusBuf
+{
+    /** \brief Encryption status. VA_ENCRYPTION_STATUS_SUCCESSFUL if 
+     *  hardware has returned detailed inforamtion, others mean the 
+     *  CENC result is invalid  */
+    VAEncryptionStatus status;
+    /* \brief status report feed back number */
+    uint16_t           status_report_feedback_number;
+    /** \brief Buf size in bytes.  0 means buf is invalid*/
+    uint32_t           buf_size;
+    /** \brief Buffer formatted as raw data from bitstream for sequence parameter,
+     *  picture parameter, SEI parameters. Or \c NULL means buf is invalid.*/
+    void               *buf;
+    /** \brief Slice buffer type, see*/
+    VACencSliceBufType slice_buf_type;
+    /** \brief Slice buffer size in bytes. 0 means slice_buf is invalid*/
+    uint32_t           slice_buf_size;
+    /** \brief Slice buffer, parsed slice header information. Or \c NULL
+     *  means slice_buf is invalid.*/
+    void               *slice_buf;
```

Comment:
I'm not quite sure I understand why there are separate `buf` and `slice_buf` elements here.  Won't you always want exactly one of them?  (Since you get an array of `VACencStatusBuf` from `vaQueryCenc()`, I would expect each one to contain a single bitstream element - either a raw decrypted header or one of the parameter structures above.)